pkgs/by-name/co/construct/package.nix

```
@@ -0,0 +1,41 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, boost
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "construct";
+  version = "master";
+
+  src = fetchFromGitHub {
+    owner = "Thomas-de-Bock";
+    repo = "${finalAttrs.pname}";
+    rev = "b109a7f67253364af2c405506a461935c6c887b6";
+    hash = "sha256-pOoqjtFkvMtof+YjV47JPu5tw/U6CU/MpL+C1Z1pgrE=";
+  };
+
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace Makefile \
+        --replace g++ c++
+  '';
+
+  buildInputs = [ boost ];
+
+  makeTarget = "main";
+
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/
+    cp -r bin $out/
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Construct is an abstraction over x86 NASM Assembly";
+    longDescription = "Construct adds features such as while loops, if statements, scoped macros and  function-call syntax to NASM Assembly.";
```

Minor nit: Remove superfluous blank space
```suggestion
    longDescription = "Construct adds features such as while loops, if statements, scoped macros and function-call syntax to NASM Assembly.";
```